department-of-veterans-affairs/caseflow
# APPEALS-24384: Searchable Dropdown click to close UI fix

Resolves [APPEALS-24384](https://jira.devops.va.gov/browse/APPEALS-24384)

# Description
Fixed a bug where a click outside of the dropdown that was not on the scrollbar would not close the searchable dropdown.

## Acceptance Criteria
- [x] Clicking outside the dropdown will correctly close the dropdown
- [x] Clicking on the scrollbar for long dropdown options does not close the dropdown
- [x] Escape still closes the dropdown

## Testing Plan
1. Switch to a user that can look at the case details for a task
2. Go to the case details for a task
3. Click the task action dropdown
4. Once the dropdown is open, click somewhere outside the dropdown that isn't the scrollbar.
5. Verify that the dropdown is closed
6. Repeat the steps above and verify that escape also closes the dropdown
7. Repeat the steps above and verify that clicking on the window scrollbar does not close the dropdown.


# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---

## Storybook Story
*For Frontend (Presentationa) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
